api_tests/registries_moderation/test_submissions.py

```diff
@@ -561,3 +563,120 @@ def test_post_force_withdraw_action_persists_comment(self, app, registration, mo
+            node.confirm_spam(save=True)
+            node.confirm_ham(save=True)
+
+            # register the project
+            if is_embargo:
+                one_month_from_now = datetime.datetime.now() + datetime.timedelta(seconds=1)
+                embargo = EmbargoFactory(end_date=one_month_from_now, user=user)
+                registration = embargo.target_registration
+                registration.provider = provider
+                registration.update_moderation_state()
+                registration.save()
+            else:
+                registration = RegistrationFactory(provider=provider, project=node, creator=user)
+
+            registration.confirm_spam(save=True)
+            registration.confirm_ham(save=True)
+
+            # approve the project
+            registration.require_approval(user=registration.creator)
+            registration.registration_approval.accept()
+            registration.refresh_from_db()
+            if is_embargo:
+                assert registration.moderation_state == RegistrationModerationStates.INITIAL.db_name
+            else:
+                assert registration.moderation_state == RegistrationModerationStates.PENDING.db_name
+
+            if is_embargo:
+                registration.sanction.accept()
+                registration.refresh_from_db()
+                assert registration.moderation_state == RegistrationModerationStates.PENDING.db_name
+
+            # approve in moderation
+            actions_payload_base['data']['attributes']['trigger'] = RegistrationModerationTriggers.ACCEPT_SUBMISSION.db_name
+            actions_payload_base['data']['attributes']['comment'] = 'The best registration Ive ever seen'
+            actions_payload_base['data']['relationships']['target']['data']['id'] = registration._id
+
+            resp = app.post_json_api(registration_actions_url, actions_payload_base, auth=moderator.auth)
+            assert resp.status_code == 201
+            assert resp.json['data']['attributes']['trigger'] == RegistrationModerationTriggers.ACCEPT_SUBMISSION.db_name
+            registration.refresh_from_db()
+
+            if is_embargo:
+                assert registration.moderation_state == RegistrationModerationStates.EMBARGO.db_name
+            else:
+                assert registration.moderation_state == RegistrationModerationStates.ACCEPTED.db_name
+
+            assert registration.is_public == should_be_public
+
+        run_subtest(
+            privacy='private',
+            is_embargo=True,
+            should_be_public=False
+        )
+        run_subtest(
+            privacy='public',
+            is_embargo=True,
+            should_be_public=False
+        )
+        run_subtest(
+            privacy='private',
+            is_embargo=False,
+            should_be_public=True
+        )
+        run_subtest(
+            privacy='public',
+            is_embargo=False,
+            should_be_public=True
+        )
```

Comment:
I appreciate the attempt to be DRY, but we prefer our tests to be individual and atomic, rather than combining into one test case. Please split this up into four tests, and don't worry about being DRY with it. No need for `if` statements in a test.